mr/src/main/java/org/apache/iceberg/mr/mapred/IcebergObjectInspectorGenerator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr.mapred;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.hadoop.hive.serde2.SerDeException;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.typeinfo.ListTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.MapTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.StructTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Types;
+
+class IcebergObjectInspectorGenerator {
+
+  protected ObjectInspector createObjectInspector(Schema schema) throws Exception {
+    List<String> columnNames = setColumnNames(schema);
+    List<TypeInfo> columnTypes = IcebergSchemaToTypeInfo.getColumnTypes(schema);
+
+    List<ObjectInspector> columnOIs = new ArrayList<>(columnTypes.size());
+    for (int i = 0; i < columnTypes.size(); i++) {
+      columnOIs.add(createObjectInspectorWorker(columnTypes.get(i)));
+    }
+    return ObjectInspectorFactory.getStandardStructObjectInspector(columnNames, columnOIs, null);
+  }
+
+  protected ObjectInspector createObjectInspectorWorker(TypeInfo typeInfo) throws Exception {
+    ObjectInspector.Category typeCategory = typeInfo.getCategory();
+
+    switch (typeCategory) {
+      case PRIMITIVE:
+        PrimitiveTypeInfo pti = (PrimitiveTypeInfo) typeInfo;
+        return PrimitiveObjectInspectorFactory.getPrimitiveJavaObjectInspector(pti);
```

Comment:
Hive uses `java.sql.Date` and `java.sql.Timestamp` so I've been working on doing some converting between types to stop Hive complaining :')

Reply:
It sounds like @guilload can help out with some of the object inspector questions.

Reply:
Maybe this can be a visitor?  We recently built a visitor the other way round Hive -> Iceberg. https://github.com/linkedin/iceberg/blob/master/hive/src/main/java/org/apache/iceberg/hive/legacy/HiveTypeToIcebergType.java